File: app/src/main/java/com/example/steaminvestmentmanager/ItemsUpdatingThread.java
package com.example.steaminvestmentmanager;

import com.example.steaminvestmentmanager.utilclasses.*;

public class ItemsUpdatingThread extends Thread {

    public ItemsUpdatingThread() {}

    @Override
    public void run() {
        while (!isInterrupted()) {
            SteamItem[] steamItems = MainActivity.getSteamItems();
            if (steamItems != null) {
                for (SteamItem steamItem : steamItems) {
                    if (steamItem.getItemIcon() == null) {
                        steamItem.updateIcon();
                    }
                    steamItem.updateItemPrice();
                    ItemListUpdater.updateSteamItemAdapter();
                }
            }
            try {
                Thread.sleep(20000);
            }catch (Exception e) {}
        }
    }
}

File: app/src/main/java/com/example/steaminvestmentmanager/ItemsUpdatingThread.kt
package com.example.steaminvestmentmanager

import com.example.steaminvestmentmanager.MainActivity.Companion.getSteamItems
import com.example.steaminvestmentmanager.utilclasses.ItemListUpdater

class ItemsUpdatingThread : Thread() {
    private var active = true
    override fun run() {
        while (!isInterrupted) {
            val steamItems = getSteamItems()
            for (steamItem in steamItems) {
                while (!active) sleep(500)
                if (steamItem.itemIcon == null) {
                    steamItem.updateIcon()
                }

                steamItem.updateItemPrice()
                ItemListUpdater.updateSteamItemAdapter()
            }
            try {
                sleep(20000)
            } catch (_: Exception) {
            }
        }
    }

    fun pauseThread() {
        active = false
    }

    fun resumeThread() {
        active = true
    }
}